finetuner/tuner/keras/__init__.py

```diff
@@ -196,8 +201,24 @@ def save(self, *args, **kwargs):
         :param kwargs: Keyword arguments to pass to ``save`` method of the embedding
             model
         """
-
+        state = {
+            'epoch': kwargs.pop('epoch', 0),
+            'best': kwargs.pop('best', False),
+            'monitor': kwargs.pop('monitor', 'train_loss'),
+        }
         self.embed_model.save(*args, **kwargs)
+        with open(
+            os.path.join(kwargs.get('filepath', './'), 'saved_state.pkl'), 'wb'
+        ) as f:
+            pickle.dump(state, f)
+
+    def load(self, fp, *args, **kwargs):
+        """Loads the embedding model, optimizer and updates the state epoch."""
+
+        self._embed_model = keras.models.load_model(fp, *args, **kwargs)
+        with open(os.path.join(fp, 'saved_state.pkl'), 'rb') as f:
+            loaded_state = pickle.load(f)
+        self.state.epoch = loaded_state['epoch']
```

Comment:
Why do we need to change this? `save` is meant to be for model only, but here we are saving the whole trainer state?